test/transcoding/atomics.spt

```diff
@@ -0,0 +1,95 @@
+3 FunctionParameter 4 7 
+3 FunctionParameter 4 8 
+3 FunctionParameter 4 9 
+3 FunctionParameter 3 10 
+
+2 Label 11 
+6 AtomicIIncrement 3 12 7 13 14 
+7 AtomicSMax 3 15 7 13 14 16 
+7 AtomicUMax 3 17 7 13 14 16 
+7 AtomicSMin 3 18 7 13 14 16 
+7 AtomicUMin 3 19 7 13 14 16 
+7 AtomicIAdd 3 20 7 13 14 13 
+7 AtomicISub 3 21 7 13 14 13 
+7 AtomicOr 3 22 7 13 14 13 
+7 AtomicXor 3 23 7 13 14 13 
+7 AtomicAnd 3 24 7 13 14 13 
+9 AtomicCompareExchange 3 25 7 13 14 14 13 16 
+7 AtomicExchange 3 26 7 13 14 13 
+6 AtomicLoad 3 27 8 13 14 
+5 AtomicStore 8 13 14 10 
+6 AtomicFlagTestAndSet 28 29 8 13 14 
+4 AtomicFlagClear 8 13 14 
+1 Return 
+
+1 FunctionEnd 
+
+
+; RUN: llvm-spirv %s -to-binary -o %t1.spv
+; RUN: spirv-val %t1.spv
+; RUN: llvm-spirv -r %t1.spv -o %t1.bc -spirv-ocl-builtins-version="CL1.2"
+; RUN: llvm-dis < %t1.bc | FileCheck %s --check-prefix=CHECK-LLVM-12
+
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_incPU3AS1Vi(i32 addrspace(1)* %dst) [[attr:#[0-9]+]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_maxPU3AS1Vii(i32 addrspace(1)* %dst, i32 0) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_maxPU3AS1Vii(i32 addrspace(1)* %dst, i32 0) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_minPU3AS1Vii(i32 addrspace(1)* %dst, i32 0) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_minPU3AS1Vii(i32 addrspace(1)* %dst, i32 0) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_addPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_subPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z9atomic_orPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_xorPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_andPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z14atomic_cmpxchgPU3AS1Viii(i32 addrspace(1)* %dst, i32 0, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z11atomic_xchgPU3AS1Vii(i32 addrspace(1)* %dst, i32 1) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z10atomic_addPU3AS1Vii(i32 addrspace(1)* %object, i32 0) [[attr]]
+; CHECK-LLVM-12: call spir_func i32 @_Z11atomic_xchgPU3AS1Vii(i32 addrspace(1)* %object, i32 %desired) [[attr]]
+; CHECK-LLVM-12: %.tmp = call spir_func i32 @_Z11atomic_xchgPU3AS1Vii(i32 addrspace(1)* %object, i32 1) [[attr]]
+; CHECK-LLVM-12: trunc i32 %.tmp to i1
+; CHECK-LLVM-12: call spir_func i32 @_Z11atomic_xchgPU3AS1Vii(i32 addrspace(1)* %object, i32 0) [[attr]]
+
+; RUN: llvm-spirv %s -to-binary -o %t2.spv
+; RUN: spirv-val %t2.spv
+; RUN: llvm-spirv -r %t2.spv -o %t2.bc -spirv-ocl-builtins-version="CL2.0"
+; RUN: llvm-dis < %t2.bc | FileCheck %s --check-prefix=CHECK-LLVM-20
+
+; CHECK-LLVM-20: call spir_func i32 @_Z25atomic_fetch_add_explicitPU3AS4VU7_Atomicii12memory_order12memory_scope(i32 addrspace(4)* %dst.as, i32 1, i32 2, i32 5) [[attr:#[0-9]+]]
```

Comment:
This strange that the last two arguments are 2, 5, while for other calls they are 5, 2. Please check.

Reply:
Fixed.